### PR TITLE
Added Validation to core/board.SetPlacements

### PR DESCRIPTION
**Pull request by TrevorPeters**
Also, added a method called GetStoneGroup that returns a Slice of Points representing the stone group, and a Boolean  for if the group is captured.

### CODE REVIEW

File: core/board/board.go [hunk 1]
```diff
@@ -170,11 +179,27 @@ func (b *Board) getNeighbors(pt *point.Point) []*point.Point {
...
-	// TODO(kashomon): Validate we have a valid board position -- i.e., one
+
+	// Validate we have a valid board position -- i.e., one
 	// without captures lying on the board.
+	expanded := make(map[point.Point]bool)
+	for _, m := range ml {
+		pt := m.Point()
+
+		if !expanded[*pt] {
+			stoneGroup, Captured := b.getStoneGroup(pt)
```
Comment (artasparks) [inline core/board/board.go]: Captured => captured (lowercase)

File: core/board/board.go [hunk 2]
```diff
@@ -170,11 +179,27 @@ func (b *Board) getNeighbors(pt *point.Point) []*point.Point {
...
 func (b *Board) SetPlacements(ml move.List) error {
+
 	for _, m := range ml {
 		b.setColor(m)
 	}
-	// TODO(kashomon): Validate we have a valid board position -- i.e., one
+
+	// Validate we have a valid board position -- i.e., one
 	// without captures lying on the board.
+	expanded := make(map[point.Point]bool)
```
Comment (artasparks) [inline core/board/board.go]: expanded => explored?

**Review by artasparks — COMMENTED** [review]
couple minor style comments

File: core/board/board.go (hunk 1 above)
Comment (TrevorPeters) [reply core/board/board.go]: oh, whoops

**Review by artasparks — APPROVED** [review]

**Pull request merged**